Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/checkins/common/CheckInRepositoryExtension.kt

```diff
@@ -0,0 +1,13 @@
+package de.rki.coronawarnapp.presencetracing.checkins.common
+
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.eventregistration.checkins.CheckInRepository
+import kotlinx.coroutines.flow.map
+
+/**
+ * Returns completed [CheckIn]s only
+ */
+val CheckInRepository.completedCheckIns
```

Comment:
couldn't we just add this to the `CheckInRepository` does it need to be an extension function?

Reply:
I like this pattern.
It's easy to test, and the Repository itself does not need to know about every extra usecase, keeping the API it offers slim.
Having many flows in a repository, makes it difficult for other tests too, as we would have to mock multiple flows, could be subscribed too, instead of just mocking a single flow (and the extension function does not need to be mocked).

Reply:
hmm true, i was mainly concerned about fragmenting responsibilities into too many places and making it hard to figure out where an implementation is coming from, but i guess ides should help with that 😄